providers/apis/defi/raydium/provider.go

```diff
@@ -0,0 +1,253 @@
+package raydium
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	binary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/programs/token"
+	"github.com/gagliardetto/solana-go/rpc"
+	"go.uber.org/zap"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	oracletypes "github.com/skip-mev/slinky/oracle/types"
+	"github.com/skip-mev/slinky/pkg/math"
+	"github.com/skip-mev/slinky/providers/base/api/handlers"
+	providertypes "github.com/skip-mev/slinky/providers/types"
+	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
+)
+
+var _ handlers.APIFetcher[mmtypes.Ticker, *big.Int] = (*APIPriceFetcher)(nil)
```

Comment:
```suggestion
var _ types.PriceAPIFetcher = (*APIPriceFetcher)(nil)
```
we already have a type alias for this.